test: add filters to save test time for kernel tests

hakehuang opened the pull request:
some platform does not have enough SRAM
so use filter to skip the build check.

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

nashif commented on tests/kernel/mem_protect/userspace/testcase.yaml:
```diff
@@ -1,6 +1,6 @@
 tests:
   kernel.memory_protection.userspace:
-    filter: CONFIG_ARCH_HAS_USERSPACE
+    filter: CONFIG_ARCH_HAS_USERSPACE and CONFIG_SRAM_SIZE > 32
```
we have min_ram and min_flash keys that can be used instead of Kconfig variables, please use those.

hakehuang replied:
ok

hakehuang committed:
test: add filters to save test time

some platform does not have enough SRAM
so use filter to skip the build check.

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

MaureenHelm reviewed (approved):
Please rebase

nashif commented on tests/kernel/sched/schedule_api/testcase.yaml:
```diff
@@ -3,19 +3,23 @@ tests:
     filter: not CONFIG_SCHED_MULTIQ
     extra_configs:
       - CONFIG_TIMESLICING=y
+    min_ram: 36
```
I think this is very restrictive, there are now at least ~50 platforms with 32kb or less that would be excluded based on this.

hakehuang replied:
any suggestion on how to handle this?

nashif commented:
> add filters to save test time for kernel tests

what time are we try to save here exactly?

hakehuang commented:
> > add filters to save test time for kernel tests
> 
> what time are we try to save here exactly?

@nashif if there is no such filter sanitcheck will try to build this application and check whether it can pass or not.
